src/Language/PureScript/CodeGen/JS.hs

```diff
@@ -113,19 +110,18 @@ moduleToJs (Module _ coms mn _ imps exps reExps foreigns decls) foreignInclude =
     maybePureGen alreadyAnnotated = \case
       AST.VariableIntroduction ss name j -> Just (AST.VariableIntroduction ss name (annotateOrWrap <$> j))
       AST.App ss f args -> (if alreadyAnnotated then AST.App else pureApp) ss <$> maybePure' f <*> traverse maybePure args
-      -- In general, indexers can be effectful, but not when indexing into an
-      -- ES module object.
-      AST.Indexer ss idx v@(AST.Var _ name)
-        | name `S.member` moduleObjectNames -> (\idx' -> AST.Indexer ss idx' v) <$> maybePure idx
       AST.ArrayLiteral ss jss -> AST.ArrayLiteral ss <$> traverse maybePure jss
       AST.ObjectLiteral ss props -> AST.ObjectLiteral ss <$> traverse (traverse maybePure) props
       AST.Comment c js -> AST.Comment c <$> maybePure js
 
+      js@(AST.Indexer _ _ (AST.Var _ "$foreign")) -> Just js
```

Comment:
Could the text `$foreign` be defined as a constant somewhere, so that a change in one place doesn't break this somewhere else?

Reply:
Done.